lib/crawler.js

```diff
@@ -377,14 +377,18 @@ class Crawler {
         throw new Error('Delete without metadata');
       }
       request.document._metadata = request.response._metadataTemplate;
+      if (request.context.history) {
+        request.document._metadata.rootRequest = request.context.history[0];
+      }
       return Q(request);
     }
 
     const metadata = {
       type: request.type,
       url: request.url,
       fetchedAt: moment.utc().toISOString(),
-      links: {}
+      links: {},
+      rootRequest: request.context.history ? request.context.history[0] : undefined,
```

Comment:
For my own education: undefined is similar to null in node?

Reply:
It's different.